### PR TITLE
#1474 Highlight Text does not provide a message that nothing has changed

### PR DESCRIPTION
**Pull request by junkiattan**
#1474 

Hey @initialshl ,

I have managed to find the spot to input the RefreshRibbonControl method. I also modified the HighlightTextEnabledHandler to check for the proper selection type and length during each refresh.
I have tested it on PowerPoint and it works as intended. I have also made sure to run the Test project to check for any errors and none were found.

Thank you for your advice earlier! ;)

### CODE REVIEW

File: PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs [hunk 1]
```diff
@@ -10,7 +13,22 @@ class HighlightTextEnabledHandler : EnabledHandler
...
+            try
+            {
+                if (this.GetAddIn().Application.ActiveWindow.Selection.Type == PpSelectionType.ppSelectionText && 
+                    this.GetAddIn().Application.ActiveWindow.Selection.TextRange2.TrimText().Length > 0)
+                {
+                    return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+                }
+                else
+                {
+                    return !this.GetRibbonUi().HighlightTextFragmentsEnabled;
```
Comment (initialshl) [inline PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs]: Is this working as intended? If HighlightTextFragmentsEnabled is false and selectiontextlength is 0, is the button incorrectly enabled?

Comment (junkiattan) [reply PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs]: I have changed it;)

File: PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs [hunk 2]
```diff
@@ -10,7 +13,22 @@ class HighlightTextEnabledHandler : EnabledHandler
...
+                    this.GetAddIn().Application.ActiveWindow.Selection.TextRange2.TrimText().Length > 0)
+                {
+                    return this.GetRibbonUi().HighlightTextFragmentsEnabled;
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (System.Runtime.InteropServices.COMException) // If this exception is caught, it means nothing has been selected yet
```
Comment (alexfjw) [inline PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextEnabledHandler.cs]: shift the comments to a new line, see Commenting Conventions [here](https://github.com/oss-generic/process/blob/master/codingStandards/CodingStandard-CSharp.adoc)

 
--

File: PowerPointLabs/PowerPointLabs/ThisAddIn.cs [hunk 3]
```diff
@@ -959,6 +959,7 @@ private void ThisAddInSelectionChanged(PowerPoint.Selection sel)
             Ribbon.RefreshRibbonControl("ReplaceWithClipboardButton");
             Ribbon.RefreshRibbonControl("PasteIntoGroupButton");
             Ribbon.RefreshRibbonControl("AddSpotlight");
+            Ribbon.RefreshRibbonControl("HighlightTextButton"); // To grey out the "HighlightText" button whenever non-text fragment or nothing has been selected
```
Comment (alexfjw) [inline PowerPointLabs/PowerPointLabs/ThisAddIn.cs]: comments to new line

File: PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextActionHandler.cs [hunk 4]
```diff
@@ -15,18 +15,19 @@ protected override void ExecuteAction(string ribbonId)
 
             if (this.GetAddIn().Application.ActiveWindow.Selection.Type == Microsoft.Office.Interop.PowerPoint.PpSelectionType.ppSelectionShapes)
             {
-                HighlightTextFragments.userSelection = HighlightTextFragments.HighlightTextSelection.kShapeSelected;
+                HighlightTextFragments.userSelection = HighlightTextFragments.HighlightTextSelection.kShapeSelected;                
             }
             else if (this.GetAddIn().Application.ActiveWindow.Selection.Type == Microsoft.Office.Interop.PowerPoint.PpSelectionType.ppSelectionText)
             {
-                HighlightTextFragments.userSelection = HighlightTextFragments.HighlightTextSelection.kTextSelected;
+                HighlightTextFragments.userSelection = HighlightTextFragments.HighlightTextSelection.kTextSelected;
```
Comment (alexfjw) [inline PowerPointLabs/PowerPointLabs/ActionFramework/HighlightLab/HighlightText/HighlightTextActionHandler.cs]: Not sure why there's whitespace added.
Might want to check your ide as it seemed to have happened in another of your PRs

**Comment by alexfjw** [comment]
See comments on style. Otherwise, code works as intended.

**Comment by junkiattan** [comment]
@alexfjw Thanx for the checks man! Yeah I have no idea what happened with regards to the whitespaces cause I dun like to leave trailing whitespaces in my code. Will take note of the Commenting Conventions!
